device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceTwin/DeviceTwin.java

```diff
@@ -90,6 +94,14 @@ public IotHubMessageResult execute(Message message, Object callbackContext)
                         if (iotHubStatus == IotHubStatusCode.OK)
                         {
                             TwinState twinState = TwinState.createFromPropertiesJson(new String(dtMessage.getBytes(), Message.DEFAULT_IOTHUB_MESSAGE_CHARSET));
+
+                            if (dtMessage.getClientPropertiesCallback() != null)
+                            {
+                                ClientPropertiesCallback propCallBack = dtMessage.getClientPropertiesCallback();
+                                Object propCallBackContext = dtMessage.getClientPropertiesCallbackContext();
```

Comment:
You guard against getClientPropertiesCallback. Do we need to guard against getClientPropertiesCallbackContext?

Reply:
In this case no, since the callback will be implemented by the customer. I'll make sure to add some documentation in the callback class.